Admin user

blueJayWays opened the pull request:
## Summary

- Create endpoint to create new admin users

- Add tests for create admin-user endpoint


## Trello Card

https://trello.com/c/fY3PqeYy/17-usuario-administrador

blueJayWays committed:
Merge sign-in into list-users

blueJayWays committed:
Add middleware to check if user is authenticated

blueJayWays committed:
Add endpoint to get a list of users

blueJayWays committed:
Add test cases for endpoint to get all users

blueJayWays committed:
Create endpoint to create new admin users

blueJayWays committed:
Add tests for create admin-user endpoint

blueJayWays committed:
Merge master into list-users

blueJayWays committed:
Merge list-user into admin-user

igna92ts commented on app/controllers/user_controller.js:
```diff
@@ -19,14 +19,73 @@ exports.signUp = (req, res, next) => {
+exports.signIn = (req, res, next) => {
   const { email, password } = req.body;
   return User.findOne({ where: { email } }).then(user => {
     if (!user) next(error.unAuthorizedError('Invalid credentials'));
```
its the same error both times, just do `if (!user || !user.validatePassword(password)`

igna92ts commented on app/controllers/user_controller.js:
```diff
@@ -19,14 +19,73 @@ exports.signUp = (req, res, next) => {
+        .update({
+          isAdmin: true
+        })
+        .then(() => {
```
no need for the {} here, just remove them and move the next line up

igna92ts commented on app/controllers/user_controller.js:
```diff
@@ -19,14 +19,73 @@ exports.signUp = (req, res, next) => {
+        });
+    } else {
+      User.create({ name, lastname, email, password, isAdmin: true })
+        .then(createdUser => {
```
same

igna92ts commented on app/controllers/user_controller.js:
```diff
@@ -19,14 +19,73 @@ exports.signUp = (req, res, next) => {
+            .json({ createdUser })
+            .end();
+        })
+        .catch(err => {
```
same

igna92ts commented on app/routes.js:
```diff
@@ -1,7 +1,21 @@
+  app.get('/users', isAuthenticated, userController.getAllUsers);
   app.post('/users', validationResultHandler(signUp), userController.signUp);
   app.post('/users/sessions', validationResultHandler(signIn), userController.signIn);
+  app.post(
```
route should be the other way around, first users and then admin `/users/admin`

igna92ts commented on test/user.js:
```diff
@@ -4,7 +4,31 @@ const chai = require('chai'),
+  return chai
+    .request(app)
+    .post('/users/sessions')
+    .send({ email: usr.email, password: usr.password });
```
properties are called the same so you can just do, `.send(usr)`

igna92ts commented on test/user.js:
```diff
@@ -4,7 +4,31 @@ const chai = require('chai'),
+  const cookies = res.headers['set-cookie'];
+  let token;
+  if (cookies) {
+    cookies.forEach(function(icookie) {
```
use arrow functions

igna92ts commented on test/user.js:
```diff
@@ -258,3 +277,173 @@ describe('/users/sessions POST', () => {
+          .end(function(err, response) {
+            expect(response).to.have.status(200);
+            response.should.be.json;
+            response.body.should.be.a('object');
```
dont mix should and expect, use one or the other.

blueJayWays committed:
Add endpoint to get a list of users

Add test cases for endpoint to get all users

Put validationResultHandler in middlewares

Merge branch 'sign-up' into test-documentation

Add test to count the users in the DB

Update docs

Merge pull request #2 from wolox-training/sign-up

Sign up
Merge pull request #4 from wolox-training/test-documentation

Test documentation
Merge pull request #5 from wolox-training/sign-in

Sign in
Merge master into list-users

blueJayWays committed:
Apply corrections indicated in PR #6

blueJayWays committed:
Update tests. Don't use 'should' instead use 'expect'

blueJayWays committed:
Merge list-users into admin-user

blueJayWays committed:
Modify pagination in get all users endpoint

blueJayWays committed:
Remove test to check of response is a object

blueJayWays committed:
Merge list-user in admin-user
